app/controllers/katello/api/v2/host_subscriptions_controller.rb

```diff
@@ -43,6 +44,40 @@ def destroy
       respond_for_destroy(:resource => @host)
     end
 
+    api :POST, "/hosts/subscriptions/", N_("Register a host with subscription and information.")
+    param :name, String, :desc => N_("Name of the host"), :required => true
+    param :uuid, String, :desc => N_("UUID to use for registered host")
```

Comment:
'setting a uuid' isn't required for connecting with a sub-man host later as you can specify whatever gets generated. I updated it to specify that it will be auto generated.